politeiawww/paywall.go

```diff
@@ -1,47 +1,280 @@
 package main
 
 import (
-	www "github.com/decred/politeia/politeiawww/api/v1"
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "github.com/decred/politeia/politeiawww/api/v1"
 	"github.com/decred/politeia/politeiawww/database"
 	"github.com/decred/politeia/util"
 )
 
-// ProcessVerifyUserPaymentTx verifies that the provided transaction
+type paywallInfo struct {
+	address     string
+	amount      uint64
+	txNotBefore int64
+}
+
+const (
+	// paywallExpiry is the amount of time the server will watch a paywall address
+	// for transactions. It gets reset when the user logs in or makes a call to
+	// RouteVerifyUserPayment.
+	paywallExpiry = time.Hour * 24
+
+	// paywallCheckGap is the amount of time the server sleeps after polling for
+	// a paywall address.
+	paywallCheckGap = time.Second * 1
+)
+
+var (
+	paywallUsers map[uint64]paywallInfo
```

Comment:
All this stuff needs to go in the context and not be global. Also use the mutex that already exists in the context instead of making a new one.